### ISSUE TITLE
Bug: Arguments precedence inconsistency in CLI `run` command

### ISSUE THREAD

**Issue by kedod** [litestar-org/litestar#3188]
### Description

For most of the options env file has higher priority but there are some for which CLI options are "superior".

```python
    reload_dirs = env.reload_dirs or reload_dir
    reload_include = env.reload_include or reload_include
    reload_exclude = env.reload_exclude or reload_exclude
    ...
    ssl_certfile = ssl_certfile or env.certfile_path
    ssl_keyfile = ssl_keyfile or env.keyfile_path
    create_self_signed_cert = create_self_signed_cert or env.create_self_signed_cert
```

In the name of consistency, this needs to be fixed!
Also CLI > ENV

### Litestar Version

2.7.0

### Platform

- [ ] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3188">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3188/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3188/pledge.svg">
</picture>
</a>

**Comment by JacobCoffee** [comment]
Oh nice catch. Would you like to open up a PR for this? I'm thinking I agree on CLI precedence here.

**Comment by provinzkraut** [comment]
Closed by #3190
